src/frontend/src/common/redux/middleware/index.ts

```diff
@@ -56,11 +61,25 @@ export const setGroupMiddleware: Middleware =
     return next(action);
   };
 
+const expirePathogenCaches = async (): Promise<void> => {
+  const queriesToRefetch = [
+    USE_PHYLO_RUN_INFO,
+    USE_SAMPLE_INFO,
+    USE_LINEAGES_INFO_QUERY_KEY,
+  ];
+
+  forEach(queriesToRefetch, async (q) => {
+    await queryClient.invalidateQueries([q]);
+    await queryClient.fetchQuery([q]);
+  });
+};
+
 export const setPathogenMiddleware: Middleware =
   () => (next) => (action: AnyAction) => {
     const { type, payload } = action;
     if (type === CZGEReduxActions.SET_PATHOGEN_ACTION_TYPE) {
       setLocalStorage(ReduxPersistenceTokens.PATHOGEN, payload);
+      expirePathogenCaches();
```

Comment:
If you've been seeing this work fine, I have no reason to think this instance will give us trouble.

That said, I think this approach could open us up to some race conditions in certain edge cases. If the code that handled `SET_PATHOGEN_ACTION_TYPE` in the reducer happened to be very slow -- say it does a lot of computation in the reducer from the payload and current state before coming to a result that then gets placed in state -- we could theoretically have a situation where the query cache gets expired and the query refetches data, but while the underlying state is still stale (b/c reducer still running b/c slow), so the new query is doing a fetch for stale data. The code feels sequential: expire cache in middleware, set new value in reducer, query fetch based on new value. But I don't think we have a real guarantee that the query fetch has to happen after the new value is set.

All that said, for this specific instance (and probably almost every case in our app as currently build), I'd be pretty shocked if it didn't run in that sequence. I expect the speed of setting a new value in redux vs doing the whole cache expiration, run query fetch is so huge we can ignore the potential race condition.

Reply:
Hmm... I do see the potential race condition.  It seems ok for now, but if there's weirdness in the future we can use thunk middleware to address.

Reply:
I don't think that this creates a race condition, per se. But almost worse, it guarantees that the caches are updated before the pathogen is set. Redux middleware executes before the action is applied to the state. One way you could get around this is to expire the caches, but to tell them which pathogen you want to fetch with when you refresh them.

https://redux.js.org/tutorials/fundamentals/part-4-store#middleware